### PR TITLE
[OpenAI] [Assistants] Adjusting customizations for newest spec release

### PR DESCRIPTION
**Pull request by jpalvarezl**
- Trying to address everything necessary for #38612 there might be still some customizations that aren't carried over.
- Regenerating from the current `HEAD` of the spec repository
- Adjusted tests making `UploadFileRequest` private
- Adjusted test recordings for non Azure
- Remaining TODO: Azure tests re-recordings [Shawn: get all recording done beside `basicRetrieval` tests in Azure Client]

- [Shawn] Manually added the  `@QueryParam("api-version") String apiVersion,` to REST client and be able to test Azure Resource. Which should be fixed from TypeSpec Emitter side. But keep it here for testing Azure Recording.

### CODE REVIEW

File: sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsClientBuilder.java [hunk 1]
```diff
@@ -343,7 +311,6 @@ private boolean useNonAzureOpenAIService() {
      *
      * @return an instance of AssistantsAsyncClient.
      */
-    @Generated
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsClientBuilder.java:346]: should not remove this genearted

File: sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureRetrievalAsyncTest.java [hunk 2]
```diff
@@ -32,14 +34,15 @@ public class AzureRetrievalAsyncTest extends AssistantsClientTestBase {
 
     AssistantsAsyncClient client;
 
+    @Disabled("Retrieval tools are not supported in Azure")
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureRetrievalAsyncTest.java:37]: @jpalvarezl  I added Disabled annotation only to `basicRetrieval` test when testing in AZURE resource

**Comment by mssfang** [comment]
As we can see all tests passed but only left Analyze step. Currently it failed on azure core dependency but will fail at TypeSpec Verification step when Core versioning issue fixed. Because I manually add the  QueryParam values in the PR

File: sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureRetrievalAsyncTest.java (hunk 2 above)
Comment (jpalvarezl) [reply sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureRetrievalAsyncTest.java:37]: Yikes! That's unfortunate, but it wouldn't be the first test we had to disable for this reason. Thanks @mssfang !

**Pull request merged**